docs/codeql/ql-language-reference/types.rst

```diff
@@ -382,6 +384,68 @@ from ``OneTwoThree`` and ``int``.
    must :ref:`override <overriding-member-predicates>` those definitions to avoid ambiguity.
    :ref:`Super expressions <super>` are often useful in this situation.
 
+.. _final-extensions:
+
+Final extensions
```

Comment:
Please add an example of a final extension of a class with a field.

Reply:
I think it would be better to do that separately, since we don't currently document overriding of fields (although we do mention it). I think it would be confusing to have an example of field shadowing here without an example of field overriding before.